ios/RNShopify.m

```diff
@@ -0,0 +1,320 @@
+{
+    [self.client getProductsPage:page completion:^(NSArray<BUYProduct *> *products, NSUInteger page, BOOL reachedEnd, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([self getDictionariesForProducts:products]);
+    }];
+}
+
+RCT_EXPORT_METHOD(getProductsWithTags:(NSUInteger)page tags:(NSArray<NSString *> *)tags resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getProductsByTags:tags page:page completion:^(NSArray<BUYProduct *> *products, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([self getDictionariesForProducts:products]);
+    }];
+}
+
+RCT_EXPORT_METHOD(getProductsWithTagsForCollection:(NSUInteger)page collectionId:(nonnull NSNumber *)collectionId tags:(NSArray<NSString *> *)tags resolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+    [self.client getProductsPage:page inCollection:collectionId withTags:tags sortOrder:BUYCollectionSortCollectionDefault completion:^(NSArray<BUYProduct *> *products, NSUInteger page, BOOL reachedEnd, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        resolve([self getDictionariesForProducts:products]);
+    }];
+}
+
+RCT_EXPORT_METHOD(webCheckout:(NSArray *)cart resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject)
+{
+    _resolve = resolve;
+    _reject = reject;
+
+    BUYCheckout *checkout = [self createCheckoutFromCart:cart];
+
+    [self.client createCheckout:checkout completion:^(BUYCheckout *checkout, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code], error.localizedDescription, error);
+        }
+
+        BUYWebCheckoutPaymentProvider *webPaymentProvider = [[BUYWebCheckoutPaymentProvider alloc] initWithClient:self.client];
+        webPaymentProvider.delegate = self;
+
+        [webPaymentProvider startCheckout:checkout];
+    }];
+}
+
+RCT_EXPORT_METHOD(checkout:(NSArray *)cart resolver:(RCTPromiseResolveBlock)resolve
+                rejecter:(RCTPromiseRejectBlock)reject)
+{
+    BUYCheckout *checkout = [self createCheckoutFromCart:cart];
+
+    [self.client createCheckout:checkout completion:^(BUYCheckout *checkout, NSError *error) {
+        if (error) {
+            return reject([NSString stringWithFormat: @"%lu", (long)error.code],
+                [self getJsonFromError:error], error);
+        }
+
+        self.checkout = checkout;
+        resolve(@"Success");
```

Comment:
Maybe a bool would be better here. Or maybe nothing at all. Can this be anything but success if the promise is resolved? It will probably be rejected in all other cases.